src/history/test/HistoryTests.cpp

```diff
@@ -713,6 +713,152 @@ TEST_CASE("History catchup with different modes",
     }
 }
 
+TEST_CASE("Long online catchup", "[history][catchup]")
+{
+    CatchupSimulation catchupSimulation{};
+
+    // Try to get to ledger 1000 with 200 ledgers buffered
+    // when there are 20 checkpoints available.
+    // In order to prevent mSyncingLedgers from growing indefinitely,
+    // it gets trimmed periodically.
+    // This test makes sure that we are able to catch up to the network
+    // even when some ledgers get trimmed from mSyncingLedgers.
+
+    auto bufferLedgers = 200;
+    auto initLedger = 1000;
+    auto numCheckpointsAvailable = 20;
+
+    auto checkpointLedger =
+        catchupSimulation.getLastCheckpointLedger(numCheckpointsAvailable);
+    catchupSimulation.ensureOnlineCatchupPossible(checkpointLedger,
+                                                  bufferLedgers);
```

Comment:
Is there a reason not to do so? The function's second argument is `bufferLedgers` so I figured it's best to pass it.

Reply:
it's redundant. there's no reason to pass it because you're already passing a checkpoint ahead of where you want to land anyway, so having the second argument is unnecessary and just makes things more confusing.

Reply:
Removed.